nix/templates/init/flake.nix

```diff
@@ -0,0 +1,104 @@
+{
+  description = "Description for the project";
+
+  # Inputs are the flake references that are used in the flake.
+  # Nix will fetch the flake and stores the hash in the flake.lock file.
+  inputs = {
+    # Nixpkgs provides the many packags that are normally available in NixOS.
+    #
+    # WARNING:
+    # We currently pin the verion to 0.2311.555610 to ensure that the zig build
+    # works correctly. Recent updates to nixpkgs did introduce breaking changes to
+    # the build environemnt variables, which makes the Zig compiler fail.
+    #
+    # Issue: https://github.com/ziglang/zig/issues/18998
+    # When resolves remove the '=' from the URL to update to the latest stable
+    # version of nixpkgs.
+    nixpkgs.url = "https://flakehub.com/f/NixOS/nixpkgs/=0.2311.555610.tar.gz";
```

Comment:
I'm currently forced to pin the version. There is a workaround to not pin the version by also requiring pkg-config. But for some reason that didn't work for me. I have some refactorings for the libpq planned, and maybe make it optional. That should solve the issue for most users.